### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by gaelL**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,73 @@
...
+
+
+  * **S3**: Amazon Simple Storage Service
+  * **IAM** Identity and Access Management policy
+
+# Requirements
+
+In order to run this task, couple elements are required within the infrastructure:
+
+  * Having an S3 bucket to store Terraform remote states [Here](https://docs.aws.amazon.com/quickstarts/latest/s3backup/step-1-create-bucket.html)
```
Comment (xlr-8) [inline README.md]:
```suggestion
  * Having an S3 bucket with versioning to store Terraform remote states [here](https://docs.aws.amazon.com/quickstarts/latest/s3backup/step-1-create-bucket.html)
```

File: README.md [hunk 2]
```diff
@@ -1,8 +1,73 @@
 # stack-amazon-s3
- Allow to create an Amazon S3 bucket
 
-<img src="https://raw.githubusercontent.com/cycloid-community-catalog/stack-amazon-s3/master/diagram.png" width="200">
+Create a Amazon S3 bucket and a dedicated user using Cycloid stacks format.
```
Comment (xlr-8) [inline README.md]:
```suggestion
Create an Amazon S3 bucket and a dedicated user using Cycloid stacks format.
```

File: pipeline/variables.sample.yml [hunk 3]
```diff
@@ -3,34 +3,69 @@
 #
 
 # Amazon access to use to run terraform. Usually admin account to be able to create Amazon services.
+
+#. aws_access_key (required): ((aws.access_key))
+#+ Amazon AWS access key for Terraform. see value format [Here](https://docs.cycloid.io/advanced-guide/integrate-and-use-cycloid-credentials-manager.html#vault-in-the-pipeline)
```
Comment (xlr-8) [inline pipeline/variables.sample.yml]:
```suggestion
#+ Amazon AWS access key for Terraform. See value format [here](https://docs.cycloid.io/advanced-guide/integrate-and-use-cycloid-credentials-manager.html#vault-in-the-pipeline).
```

File: pipeline/variables.sample.yml [hunk 4]
```diff
@@ -3,34 +3,69 @@
...
 aws_access_key: ((aws.access_key))
+
+#. aws_secret_key (required): ((aws.secret_key))
+#+ Amazon AWS secret key for Terraform. see value format [Here](https://docs.cycloid.io/advanced-guide/integrate-and-use-cycloid-credentials-manager.html#vault-in-the-pipeline)
```
Comment (xlr-8) [inline pipeline/variables.sample.yml]:
```suggestion
#+ Amazon AWS secret key for Terraform. See value format [here](https://docs.cycloid.io/advanced-guide/integrate-and-use-cycloid-credentials-manager.html#vault-in-the-pipeline).
```

**Review by xlr-8 — COMMENTED** [review]
Couple suggestions, need to regenerate the README

File: README.md [hunk 5]
```diff
@@ -1,8 +1,73 @@
...
+
+In order to run this task, couple elements are required within the infrastructure:
+
+  * Having an S3 bucket to store Terraform remote states [Here](https://docs.aws.amazon.com/quickstarts/latest/s3backup/step-1-create-bucket.html)
+
+# Details
+
+## Pipeline
+
+> **Note** The pipeline contains a manual approval between terraform plan and terraform apply.
```
Comment (sdurrheimer) [inline README.md]:
```suggestion
> **Note** The pipeline contains a manual approval between `terraform plan` and `terraform apply`.
```

File: README.md [hunk 6]
```diff
@@ -1,8 +1,73 @@
...
 > That means if you trigger a terraform plan, to apply it, you have to go on terraform apply job
```
Comment (sdurrheimer) [inline README.md]:
```suggestion
> That means if you trigger a `terraform plan`, to apply it, you have to go on `terraform apply` job
```

File: pipeline/variables.sample.yml [hunk 7]
```diff
@@ -3,34 +3,69 @@
...
 aws_secret_key: ((aws.secret_key))
+
+#. aws_default_region (required): eu-west-1
+#+ Amazon AWS region to use for Terraform.
 aws_default_region: eu-west-1
 
-# Bucket configuration to store terraform remote state file.
+#. terraform_storage_bucket_name (required): ($ organization_canonical $)-terraform-remote-state
+#+ AWS S3 bucket name to store terraform remote state file.
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml]:
```suggestion
#+ AWS S3 bucket name to store Terraform remote state file.
```

File: pipeline/variables.sample.yml [hunk 8]
```diff
@@ -3,34 +3,69 @@
...
 terraform_storage_bucket_name: ($ organization_canonical $)-terraform-remote-state
+
+#. terraform_storage_bucket_path (required): ($ project $)/($ environment $)
+#+ AWS S3 bucket path to store terraform remote state file.
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml]:
```suggestion
#+ AWS S3 bucket path to store Terraform remote state file.
```

File: pipeline/variables.sample.yml [hunk 9]
```diff
@@ -3,34 +3,69 @@
...
 terraform_storage_bucket_path: ($ project $)/($ environment $)
 
 
 #
 # Repos
 #
 
-# Configuration of the git repository used to configure the stack
+#. config_git_repository (required): git@github.com:MyUser/config-s3.git
+#+ Git repository url containing the config of the stack.
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml]:
```suggestion
#+ Git repository URL containing the config of the stack.
```

File: pipeline/variables.sample.yml [hunk 10]
```diff
@@ -3,34 +3,69 @@
...
 config_git_repository: git@github.com:MyUser/config-s3.git
+
+#. config_git_branch (required): master
+#+ Branch of the config git repository.
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml:37]:
```suggestion
#+ Branch of the config Git repository.
```

File: pipeline/variables.sample.yml [hunk 11]
```diff
@@ -3,34 +3,69 @@
...
 config_git_branch: master
+
+#. config_git_private_key (required): ((git_config.ssh_key))
+#+ SSH key pair to fetch the config git repository.
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml:41]:
```suggestion
#+ SSH key pair to fetch the config Git repository.
```

File: pipeline/variables.sample.yml [hunk 12]
```diff
@@ -3,34 +3,69 @@
...
+#+ Name of the project.
 project: ($ project $)
+
+#. customer (required): ($ organization_canonical $)
+#+ Name of the Cycloid Organization, used as customer variable name.
 customer: ($ organization_canonical $)
 
-# If you need to use a specific branch of stack
+#. stack_git_branch (required): master
+#+ Branch to use on the public stack git repository
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml:62]:
```suggestion
#+ Branch to use on the public stack Git repository.
```

File: pipeline/variables.sample.yml [hunk 13]
```diff
@@ -3,34 +3,69 @@
...
 stack_git_branch: master
 
-# Path of your configuration
+#. config_ansible_path (required): ($ project $)/ansible
+#+ Path of Ansible files in the config git repository
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml:66]:
```suggestion
#+ Path of Ansible files in the config Git repository.
```

File: pipeline/variables.sample.yml [hunk 14]
```diff
@@ -3,34 +3,69 @@
...
+config_ansible_path: ($ project $)/ansible
+
+#. config_terraform_path (required): ($ project $)/terraform/($ environment $)
+#+ Path of Terraform files in the config git repository
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml:70]:
```suggestion
#+ Path of Terraform files in the config Git repository.
```

File: terraform/outputs.tf [hunk 15]
```diff
@@ -1,11 +1,14 @@
 output "iam_s3_key" {
-  value = "${module.s3.iam_s3_key}"
+  value       = "${module.s3.iam_s3_key}"
+  description = "IAM access key dedicated to the S3 bucket access"
```
Comment (sdurrheimer) [inline terraform/outputs.tf:3]:
```suggestion
  description = "IAM access key dedicated to the S3 bucket access."
```

File: terraform/outputs.tf [hunk 16]
```diff
@@ -1,11 +1,14 @@
...
 }
 
 output "iam_s3_secret" {
-  value = "${module.s3.iam_s3_secret}"
+  value       = "${module.s3.iam_s3_secret}"
+  description = "IAM secret key dedicated to the S3 bucket access"
```
Comment (sdurrheimer) [inline terraform/outputs.tf]:
```suggestion
  description = "IAM secret key dedicated to the S3 bucket access."
```

File: terraform/outputs.tf [hunk 17]
```diff
@@ -1,11 +1,14 @@
...
 }
 
 output "s3_id" {
-  value = "${module.s3.s3_id}"
+  value       = "${module.s3.s3_id}"
+  description = "S3 bucket name"
```
Comment (sdurrheimer) [inline terraform/outputs.tf]:
```suggestion
  description = "S3 bucket name."
```

File: terraform/s3.tf.sample [hunk 18]
```diff
@@ -10,10 +10,15 @@ module "s3" {
...
   env      = "${var.env}"
   customer = "${var.customer}"
-
   #####################################
 
-  bucket_name              = "<bucket name>" #only lowercase alphanumeric characters and hyphens allowed
-  #bucket_acl = "<acl private (default) or public>"
-  #versioning_enabled = "<enable object versioning>"
+  #. bucket_name (required):
+  #+ Name of the S3 bvcket to create. Only lowercase alphanumeric characters and hyphens allowed
```
Comment (sdurrheimer) [inline terraform/s3.tf.sample]:
```suggestion
  #+ Name of the S3 bucket to create. Only lowercase alphanumeric characters and hyphens allowed.
```

File: terraform/s3.tf.sample [hunk 19]
```diff
@@ -10,10 +10,15 @@ module "s3" {
...
+  bucket_name              = "<bucket name>"
+
+  #. bucket_acl (optional): private
+  #+ S3 Canned ACL : https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl
```
Comment (sdurrheimer) [inline terraform/s3.tf.sample]:
```suggestion
  #+ S3 canned ACL: https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl
```

File: terraform/s3.tf.sample [hunk 20]
```diff
@@ -10,10 +10,15 @@ module "s3" {
...
+
+  #. versioning_enabled (optional, bool): false
+  #+ Enable S3 bucket versionning
```
Comment (sdurrheimer) [inline terraform/s3.tf.sample]:
```suggestion
  #+ Enable S3 bucket versionning.
```

**Pull request merged**